bsc.generics/generics.exp

```diff
@@ -0,0 +1,9 @@
+
```

Comment:
Should we consider put these tests in `bsc.lib/Generic/` instead of creating a new top level?  (Or even `bsc.misc/generics/` or `bsc.typechecker/generics/`, if you prefer.)

Below, there's a test for `CShow`, which is something that I would typically put under `bsc.lib/CShow/`.

Tests can be categorized in multiple ways, and unfortunately we can only put it in one directory (at the moment).  It's easy at the moment to run a group of tests in the same directory (or under the same root directory), but I'd prefer to avoid creating too many top-level directories.  (Some current ones need to go away, even.)  If it helps, there is an old mechanism in the testsuite to name a group of directories (in a `*.list` file) and run that group, so maybe if the Generic tests are multiple places, that can be used to run them all (until we move to some better testsuite).

Reply:
I wouldn't object to that, although @nanavati suggested that this should be in a new top level directory - any thoughts from him first before I move these?  

The `CShow` tests are with generics since this library was just a generics test case until recently when I moved into the bsc standard library in my bsc pull request.  But I agree this could belong separately under `bsc.lib`.

Reply:
I'd be fine with putting these tests in a subdirectory. I think I suggested the top-level because I wasn't sure which directory was appropriate (`bsc.typechecker` felt off because there were other things tangled in, `bsc.lib` because there's no Generic library to import). I didn't think of `bsc.misc` and while it feels like a bit of a cop-out I can see preferring that to a new top-level directory. 

Since we're adding CShow as part of the library I agree those tests belong in `bsc.lib` regardless.

Reply:
I notice that a few things in `bsc.lib` have moved into Prelude, so there are already tests there for things that aren't imported explicitly.  If the Generic tests were only about the typeclass, then I'd have no qualms about putting them under `bsc.lib`; but it's also testing the deriving behavior in BSC.  For that reason, I don't object to creating `bsc.typechecker/generics/` -- there's already a `deriving` directory under `bsc.typechecker` and generics (as typeclasses) is a typecheck-level thing.  What are the other things that are tangled in that made `bsc.typechecker` feel off?  (And is it any more than the other features that have subdirectories there, like don't-care?)  I'm starting to think that's the right place.